app/src/main/java/org/mozilla/fenix/utils/ScreenshotsGuardian.kt

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.utils
+
+import android.view.WindowManager
+import androidx.appcompat.app.AppCompatActivity
+import org.mozilla.fenix.ext.settings
+
+class ScreenshotsGuardian {
```

Comment:
Given it's importance and responsibility maybe some comments to document the behavior and some small tests would help.